gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -182,6 +182,7 @@ public void close()
         cleanUpWorkingDirectory();
       } finally {
         super.close();
+        fs.close();
```

Comment:
The benefit is to prevent this kind of error from happening again. As MR launcher has almost the longest life span and this can easily happen again if shared object is being closed. As MR launcher is not frequently instantiated, one per process if I understand correctly, I think the downside is minimal.